### PR TITLE
Add an executable which creates the radicle reference doc.

### LINKED ISSUE #155: Reference docs and tutorial
- [x] Radicle reference
- [x] Radicle tutorial

### PR DESCRIPTION
**Pull request by jameshaydon**
An executable which creates the radicle reference doc. This loads the radicle prelude and extracts all the variables in scope. It checks they are all documented and creates a markdown document with all the docstrings.

Executable is run on CI, ensuring that all definitions in the prelude are documented and in the reference doc.

Closes #155.

### CODE REVIEW

**Comment by geigerzaehler** [comment]
It would be great if we ran `ref-doc` in CI to ensure that nothing is undocumented.

File: exe/ReferenceDoc.hs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    checkAllDocumented undocd
+    let e = Map.fromList [ (fromIdent i, d) | (i, Just d, _) <- vars ]
+    checkAllInReference e
+    d <- Doc.cleanupPandocMd (doc e) `lPanic` "Invalid markdown."
+    writeFile "docs/source/reference.md" d
+  where
+    doc e =
+      let sec tit fns p = "## " <> tit <> "\n\n" <> p <> "\n\n" <> funs e fns in
+      T.intercalate "\n\n"
+      [ "# Radicle reference"
```
Comment (jkarni) [inline exe/ReferenceDoc.hs]: You can also use something like [interpolate](http://hackage.haskell.org/package/interpolate-0.2.0/docs/Data-String-Interpolate.html) here to make things a little more readable.

File: rad/chain.rad [hunk 2]
```diff
@@ -126,8 +127,12 @@
             mod-state2))
     (list :ok mod-state3)))
 
-(def eval__ eval)
+(def eval__ "The eval in place when `chain.rad` is loaded." eval)
+
 (def eval
+  "An eval in which one can use `(:enter-chain url)` to make the
+  eval behave as that of a remote chain, and `:send` to send all
```
Comment (jkarni) [inline rad/chain.rad]: I guess we'll leave switching all docs to this new format to a later PR?

Comment (jameshaydon) [reply rad/chain.rad]: :100: https://github.com/oscoin/radicle/issues/181

**Pull request merged**